src/SeleniumLibrary/keywords/waiting.py

```diff
@@ -135,15 +134,26 @@ def wait_until_page_contains_element(self, locator, timeout=None,
 
         ``error`` can be used to override the default error message.
         """
-        self._wait_until(
-            lambda: self.find_element(locator, required=False) is not None,
-            "Element '%s' did not appear in <TIMEOUT>." % locator,
-            timeout, error
-        )
+        if is_noney(limit):
+            self._wait_until(
+                lambda: self.find_element(locator, required=False) is not None,
+                "Element '%s' did not appear in <TIMEOUT>." % locator,
+                timeout, error
+            )
+        else:
+            limit = int(limit)
+            self._wait_until(
+                lambda: len(self.find_elements(locator)) is limit,
+                'Page should have contained "{}" element(s),'
+                'but it did contain "{}" element(s).'
+                'using locator "{}" did not appear in <TIMEOUT>.'.format(
+                    limit, len(self.find_elements(locator)), locator)
+                , timeout, error
+            )
 
     @keyword
     def wait_until_page_does_not_contain_element(self, locator, timeout=None,
-                                                 error=None):
+                                                 error=None, limit=None):
```

Comment:
Should the default value be 1? It would better describe what default functionality is and it could be easier to document. The None refers to not set and may lead to wrong assumptions.

Could you also document new argument in the keywords documentation.

Reply:
I would to keep this consistent with implementation of
Page Should Contains No Element

Reply:
True, lets use the same value for consistency reasons. Good catch.